[Analog Output]Support N sample (#23)
ni/NI-ELVIS-III-Python-Examples: docs/NI_ELVIS_III_Generating_FPGA_Clocks.md

<p align="center"><img src="../docs/resource/generatingFpgaClocks.png"/></p>

In the previous figure, you toggle the clock signal on every rising edge to generate the 20 MHz clock signal and toggle the clock signal on every other rising edge to generate the 10 MHz clock signal.

The frequency to generate on the hardware must meet the following requirements:
 - The frequency must be divisible by 25 ns because the clock cycle is 25 ns and you count the number of rising edges to make changes. For example, you cannot generate a frequency of 25 MHz. The first achievable frequency below 40 MHz is 20 MHz. 
 - The frequency must be within the range of approximately 610.35 Hz to 40 MHz. The hardware uses a 16-bit counter that counts from 0 to 65,535. You can calculate the lowest achievable frequency by the equation 1/(25 ns * 65536) ≈ 610.35 Hz. 

You can use clock divisors to generate even lower frequencies for different I/O types. You can divide the base frequency by even numbers and use the generated clock to increment the counter. For example, with a clock divisor of 2, the lowest achievable frequency is 1/(2 * 25 ns * 65536) ≈ 305.17 Hz.

> #### The following equation calculates frequencies that you can generate for pulse width modulation (PWM) I/O.

Reviewer: ```suggestion
#### The following = calculates the frequencies that you want to generate for pulse width modulation (PWM) I/O.
``` This is a test. If I left this comment successfully, you will find some words in the above sentence are highlighted.
